samples/network/manage-ip-address/Program.cs

```diff
@@ -24,38 +25,33 @@ public class Program
         //   - Get the assigned public IP address for a virtual machine
         //   - Remove a public IP address from a virtual machine.
 
-        private static readonly string UserName = "tirekicker";
-        private static readonly string Password = "<password>";
-        private static readonly string SubscriptionId = Environment.GetEnvironmentVariable("AZURE_SUBSCRIPTION_ID");
 
-        public static async Task RunSample(TokenCredential credential)
+        public static async Task RunSample()
         {
-            string publicIPAddressName1 = Utilities.RandomResourceName("pip1", 20);
-            string publicIPAddressName2 = Utilities.RandomResourceName("pip2", 20);
-            string publicIPAddressLeafDNS1 = Utilities.RandomResourceName("pip1", 20);
-            string publicIPAddressLeafDNS2 = Utilities.RandomResourceName("pip2", 20);
-            string networkInterfaceName = Utilities.RandomResourceName("nic", 20);
-            string vmName = Utilities.RandomResourceName("vm", 8);
-            string rgName = Utilities.RandomResourceName("rgNEMP", 24);
-
-            var networkManagementClient = new NetworkManagementClient(SubscriptionId, credential);
-            var virtualNetworks = networkManagementClient.VirtualNetworks;
-            var publicIPAddresses = networkManagementClient.PublicIPAddresses;
-            var networkInterfaces = networkManagementClient.NetworkInterfaces;
-            var computeManagementClient = new ComputeManagementClient(SubscriptionId, credential);
-            var virtualMachines = computeManagementClient.VirtualMachines;
-
+            const string UserName = "tirekicker";
+            const string Password = "RX-78-gp$3";
+            const string publicIPAddressName1 = "pip1";
+            const string publicIPAddressName2 = "pip2";
+            const string publicIPAddressLeafDNS1 = "pipdns1";
+            const string publicIPAddressLeafDNS2 = "pipdns2";
+            const string networkInterfaceName = "nic";
+            const string vmName = "vm";
+            const string rgName = "rgNEMP";
+
+            // create an ArmClient as the entry of all resource management API
+            ArmClient armClient = new ArmClient(new DefaultAzureCredential());
+
+            ResourceGroup resourceGroup = null;
             try
```

Comment:
what's a so big try...catch block!
A block/method should serve only one purpose.

Reply:
The purpose of this `try...catch` block is to ensure the resource group is deleted at the end of this program. But as an example (not test), it's not necessary to delete the resource group. I'll remove this `try...catch` and add a commented resource group delete statement.